src/v/security/oidc_authenticator.cc

```diff
@@ -182,6 +183,8 @@ ss::future<result<bytes>> sasl_authenticator::authenticate(bytes auth_bytes) {
     }
 
     _principal = principal_res.assume_value();
+    _audit_user.type_id = audit::user::type::user;
```

Comment:
I guess this is based off of what the rule selects, right?  It may not be the sub but it could be something else?

Maybe after #14772 gets merged, I can populate this based on which claim is being selected?

Reply:
> I guess this is based off of what the rule selects, right? It may not be the sub but it could be something else?
> 
> Maybe after #14772 gets merged, I can populate this based on which claim is being selected?

I don't really know what credential_uid represents, but I'd be tempted to set it directly to `sub` i.e., what OAuth gives us for the `openid` `scope`, and before it's mapped (which can fail).

Reply:
Sorry if my understanding is incorret, but `sub` is just one claim that a principal can be based off of, correct?  e.g. `email` is another claim?  Sorry if my terminology is incorrect

Reply:
1. [`sub`](https://www.rfc-editor.org/rfc/rfc7519#section-4.1.2) "The subject value MUST either be scoped to be locally unique in the context of the issuer or be globally unique."
2. The only required [`scope`](https://openid.net/specs/openid-connect-basic-1_0.html#Scopes) for OpenID Connect is `openid`
3. [`sub`](https://openid.net/specs/openid-connect-basic-1_0.html#Scopes:~:text=REQUIRED.%20Subject%20Identifier.) is the only required field in the token that's useful